datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -214,7 +214,11 @@ fn quarter_month(date: &NaiveDateTime) -> u32 {
     1 + 3 * ((date.month() - 1) / 3)
 }
 
-fn date_trunc_single(granularity: &str, value: i64) -> Result<i64> {
+/// Tuncates the single `value`, expressed in nanoseconds since the
```

Comment:
I think this is correct -- @Weijun-H  does it match your understanding?

Reply:
Nice simplification 👍 !